resources/compass-runtime-agent/values.yaml

```diff
@@ -39,8 +39,11 @@ compassRuntimeAgent:
         namespace: compass-system
     caCertificate:
       secret:
-        name: app-connector-certs
+        name: kyma-gateway-certs-cacert
```

Comment:
Why such name change? It implies that this is the main kyma-gateway certificate, while it's rather app-connector related cert. Isn't it?

Reply:
Some time ago `Application Connectivity Certs Sync` deployment was introduced. The responsibility of the deployment was copying TLS certificates to `app-connector-certs` secret. The implications were as follows:
- the logic for creating `app-connector-certs` secret was implemented in `Application Connectivity Certs Sync` deployment and `Certs Setup Job`
- the logic for setting `app-connector-certs` secret was in two places (some keys were set by `Application Connectivity Certs Sync` and some were set by `Certs Setup Job`)

The rename reduced an overall complexity a lot:
- `Application Connectivity Certs Sync` deployment has been removed 
- `Certs Setup Job` is the only place where `Application Connectivity Certs` are managed
- `kyma-gateway-application-connector` Gateway uses both `kyma-gateway-certs` and `kyma-gateway-certs-cacert`

Reply:
But it looks like it's incorrect.
Some thoughts, please correct me if I am wrong.
- Kyma Gateway is using `credentialName: kyma-gateway-certs` attribute
- It points to a secret with key and TLS certificate used to handle HTTPS traffic  to the cluster. It's a main "entry" for all external traffic (not related to application-connector)
- Kyma Gateway TLS mode is `SIMPLE` - it does not support Mutual TLS at this moment
- Configuring another secret with name `kyma-gateway-certs-cacert` is, according to Istio documentation, a way to provide Certificate necessary to validate client's certificates used in Mutual TLS.
- So one hand we're configuring the Kyma-Gateway to NOT use Mutual TLS (because it's mode is SIMPLE), on the other we're explicitly providing some Certificate that should be used for Mutual TLS...
- Istio may reject such configuration or treat that as mis-configuration and yield warnings, if it's not doing that at the moment
- Istio experts and other people trying to debug problems on the cluster will be certainly confused by such configuration.

If you need Mutual TLS for application-connector gateway, it should have it's own secret with a name distinct from the secret used by Kyma-Gateway.
Unless I deeply misunderstand how Mutual TLS in Istio works (it may be the case...) I think that what is proposed here is a "hack" that works only by accident - Istio should complain about misconfiguration of Kyma-Gateway.

Reply:
I have to support @Tomasz-Smelcerz-SAP in this case.

We are now under construction of multiple domains. Each new domain, which is managed by client, requires a separate Gateway. As a result, the Gateway requires reference to the correct certificate for the domain. When clients are going to manage their domains with wildcards certificates there would always be a pair - Gateway and a Secret that comes from a Certificate. If something goes wrong with a network traffic it should be traceable which certificate fails. The approach that you introduce would suggest that generic Kyma-gateway has invalid cert while it is not true - it is your application connector. Whose responsibility would it be then to debug such problems? I have mixed feelings about this approach and wondered if there is some other way that can be taken instead.

Reply:
Some facts:
- This PR reverts changes done in [this PR](https://github.com/kyma-project/kyma/pull/9845) as a fix for [this issue](https://github.com/kyma-project/kyma/issues/9795).
- According to the [discussion](https://github.com/kyma-project/kyma/issues/9795#issuecomment-719539250) in #9795 the secret was renamed from `kyma-gateway-certs-cacert` to `app-connector-certs` since there was an issue in Istio 1.7.1. What is important, according to the comment the Istio issue has been fixed in 1.7.4.
- The solution proposed in this solution has been tested thoroughly. Moreover it was working before Istio was upgraded to 1.7.1

Let me to answer your concerns.

> - So one hand we're configuring the Kyma-Gateway to NOT use Mutual TLS (because it's mode is SIMPLE), on the other we're explicitly providing some configuration - a Certificate - that should be used in Mutual TLS setup...
> - Istio may reject such configuration or treat that as mis-configuration and yield warnings, if it's not doing that at the moment
> - Istio experts and other people trying to debug problems on the cluster will be certainly confused by such configuration.

Let's consider `kyma-gateway`. Istio documentation describes `credentialName` as follows:

> For gateways running on Kubernetes, the name of the secret that holds the TLS certs including the CA certificates. Applicable only on Kubernetes. The secret (of type generic) should contain the following keys and values: key: <privateKey> and cert: <serverCert>. For mutual TLS, cacert: <CACertificate> can be provided in the same secret or a separate secret named <secret>-cacert. Secret of type tls for server certificates along with ca.crt key for CA certificates is also supported. Only one of server certificates and CA certificate or credentialName can be specified.


We have two gateways (`kyma-gateway` and `kyma-gateway-application-connector`) that share TLS certificates. I think that is ok.
As it comes to `kyma-gateway`, the type is `SIMPLE` so `kyma-gateway-certs-cacert` is simply ignored. I cannot find any information in the Istio documentation that proves this setup is incorrect.

Reply:
> If something goes wrong with a network traffic it should be traceable which certificate fails. The approach that you introduce would suggest that generic Kyma-gateway has invalid cert while it is not true - it is your application connector. Whose responsibility would it be then to debug such problems? I have mixed feelings about this approach and wondered if there is some other way that can be taken instead.

I don't get your point. 
We have the following case:
- One gateway in SIMPLE mode, one gateway in MUTUAL mode
- Both share server certificate

So if server (TLS) certificate is incorrect then all requests will fail (no matter if accessed resource is protected with TLS or mTLS ). It will be quite easy to observe as TLS handshake will fail. In case CA Cert is incorrect only requests that access mTLS protected resources will fail.

Reply:
I'm supporting @akgalwas here. Please notice following arguments:

- The only reason we have now this separate secret and deployment to synchronise certificates is long gone fixed Istio bug. If this bug hadn't happened - nobody wouldn't have separated TLS credentials for both gateways into two secrets. 

- We never had separate TLS certs for different gateways. Kyma was designed to provide same (TLS) certificate to secure them both. In the current solution it was implemented by having two secrets with SAME (TLS) certificate data manually synchronised by separate deployment. This is overhead. We can do it in much simpler way without these intermediate elements.
  
-  In the current and proposed solution both gateways end with loading SAME (TLS) certificates data to secure connection. An additional layer of security is provided by mTLS mode in `app-connector-gateway` where a separate mTLS certificate is loaded from additional secret `kyma-gateway-certs-cacert` which is exactly how it is supposed to work.

- From Istio point of view such configuration is fine. In fact it is not recommended for WWW endpoints. Using same (TLS) certificate to secure multiple gateways [can cause 404 error](https://istio.io/latest/docs/ops/common-problems/network-issues/#404-errors-occur-when-multiple-gateways-configured-with-same-tls-certificate) 
This is a reason the when running `istioctl analyze` tool we can see the warning. 
`(Gateway kyma-gateway-application-connector.kyma-system) Duplicate certificate in multiple gateways [kyma-system/kyma-gateway-application-connector kyma-system/kyma-gateway] may cause 404s if clients re-use HTTP2 connections.
` 
But we are sure that for our case (we do not use HTTP2 and we have no browser interaction on `connector-gateway`)  both gateways will work fine.
 
- If you find the naming confusing, we can consider renaming secret `kyma-gateway-certs` into something more generic like `kyma-tls-certs`. In fact this SAME (TLS) certificate is used in both gateways.